File: app/src/main/java/space/lopatkin/spb/helpboardgamecard/domain/usecase/SaveKeyboardTypeByUserChoiceUseCase.java
package space.lopatkin.spb.helpboardgamecard.domain.usecase;

import space.lopatkin.spb.helpboardgamecard.domain.model.Message;
import space.lopatkin.spb.helpboardgamecard.domain.repository.AppRepository;
import space.lopatkin.spb.helpboardgamecard.domain.model.KeyboardType;

public class SaveKeyboardTypeByUserChoiceUseCase {

    private AppRepository repository;

    public SaveKeyboardTypeByUserChoiceUseCase(AppRepository repository) {
        this.repository = repository;
    }

    public Message execute(Object userChoice) {
        if (userChoice != null) {
            KeyboardType type = KeyboardType.getOrdinalFrom(userChoice.toString());
            repository.saveKeyboardType(type.ordinal());
            return Message.ACTION_ENDED_SUCCESS;
        }
        return Message.ACTION_ENDED_ERROR;
    }

}

File: app/src/main/java/space/lopatkin/spb/helpboardgamecard/domain/usecase/SaveKeyboardTypeByUserChoiceUseCase.kt
package space.lopatkin.spb.helpboardgamecard.domain.usecase

import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.flatMapMerge
import kotlinx.coroutines.flow.flow
import space.lopatkin.spb.helpboardgamecard.domain.model.KeyboardType
import space.lopatkin.spb.helpboardgamecard.domain.repository.SettingsRepository
import space.lopatkin.spb.helpboardgamecard.domain.model.Completable

class SaveKeyboardTypeByUserChoiceUseCase(private val repository: SettingsRepository) {

    fun execute(userChoice: Any?): Flow<Completable> {
        return flow {
            if (userChoice == null) {
                throw Exception("NotFoundException (usecase): data (userChoiceKeyboardType) is null")
            } else {
                val type: KeyboardType = KeyboardType.getOrdinalFrom(userChoice.toString())
                emit(type)
            }
        }.flatMapMerge { data -> repository.saveKeyboardType(data) }
    }

}
